docs/spec/sources.md

```diff
@@ -185,9 +185,32 @@ validation:
                   on the event as an extension attribute independently."
 ```
 
-<!-- TOOD(#1550): Follow-up with a registry implementation. -->
+### Source Registry
 
-## Source RBAC
+Source CRDs SHOULD use a standard annotation to expose the types of events they can
+emit in order to ease discoverability.
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    registry.knative.dev/eventTypes: |  # <-- optional but encouraged.
+      [
+        { "type": "foo", "schema": "foo-schema", "description" : "Foo event description" },
+        { "type": "bar", "schema": "bar-schema", "description" : "Bar event description" },
+        ...
+      ]
+```
+
+If specified, the annotation SHOULD be a valid JSON array so that it can be easily unmarshalled by tooling 
```

Comment:
<!--whitespace-bot-->
Remove trailing whitespace:
```suggestion
If specified, the annotation SHOULD be a valid JSON array so that it can be easily unmarshalled by tooling
```